### PR TITLE
Add support for ShmSize to quadlet

### PR DESCRIPTION
**Pull request by rhatdan**
I am working on running android auto in a quadlet.

[Container]
AddDevice=/dev/dri/renderD128
AddDevice=/dev/kvm
DropCapability=all
Environment=PULSE_SERVER=$XDG_RUNTIME_DIR/pulse/native Environment=WAYLAND_DISPLAY=wayland-0
Environment=XDG_RUNTIME_DIR
Image=quay.io/slopezpa/qemu-aaos
ContainerName=Android
PodmanArgs=--shm-size=5g
SecurityLabelDisable=true
Volume=$XDG_RUNTIME_DIR:$XDG_RUNTIME_DIR

And I need to be able to set the --shm-size option.

#### Does this PR introduce a user-facing change?

```release-note
Podman quadlet now supports ShmSize option in unit files.
```

### CODE REVIEW

**Comment by baude** [comment]
LGTM

**Comment by rhatdan** [comment]
@vrothberg @giuseppe @flouthoc @ygalblum @containers/podman-maintainers PTAL

**Review by flouthoc — APPROVED** [review]
LGTM

File: docs/source/markdown/podman-systemd.unit.5.md
Comment (ygalblum) [inline docs/source/markdown/podman-systemd.unit.5.md:453]: We try not to copy/paste the documentation from that of `podman-run`. Instead, just say that it is the equivalent of `--shm-size`

File: pkg/systemd/quadlet/quadlet.go
Comment (ygalblum) [inline pkg/systemd/quadlet/quadlet.go:491]: The doc says that the size may be omitted, but I'm not sure this takes it into account (or at least, there's not test that checks it)

Comment (vrothberg) [reply pkg/systemd/quadlet/quadlet.go:491]: I think that's an internal Podman behavior that would need a runtime test.

File: docs/source/markdown/podman-systemd.unit.5.md
Comment (rhatdan) [reply docs/source/markdown/podman-systemd.unit.5.md:453]: I don't like forcing users to open multiple man pages to figure out how a function will work.  I guess we need to work with @edsantiago scripts to figure a way to allow podman-systemd.unit to import the same text as podman run and podman create man pages. @edsantiago What do you think?

Comment (ygalblum) [reply docs/source/markdown/podman-systemd.unit.5.md:453]: I understand your point. But, unless we use a single source of truth, maintaining the documentation will become very hard

Comment (rhatdan) [reply docs/source/markdown/podman-systemd.unit.5.md:453]: Yup,

Comment (rhatdan) [reply docs/source/markdown/podman-systemd.unit.5.md:453]: But we are pushing the pain onto the users rather then having the developers handle the pain.

Comment (ygalblum) [reply docs/source/markdown/podman-systemd.unit.5.md:453]: I agree. But, I think at this stage it's better to align to what's already there (i.e. reference to the podman command) and then solve it as a whole

Comment (edsantiago) [reply docs/source/markdown/podman-systemd.unit.5.md:453]: Well, the first thing I see here is that none of this makes any sense whatsoever.

In the man page we have:
```
--shm-size=number[unit]
...blah blah a unit can be
```
Here we have
```
ShmSize=
blah blah a unit can be     <<<<-----WTF is a "unit"? Where did "unit" come from?
```
My general sense is that this man page does not (ATM) include useful context hints like the man pages. Therefore copy-pasting (or some new TBD include mechanism) is impractical, even anti-helpful. Until that problem is solved, I suggest sticking with "see other man page".

**Comment by TomSweeneyRedHat** [comment]
LGTM once @ygalblum's concern is addressed

**Pull request merged**
